packages/contracts-ops/contracts/scripts/UpgradeCallBatches.s.sol

```diff
@@ -78,3 +55,32 @@ contract UpgradeCallBatches is Script, Config, CallBatch {
         }
     }
 }
+
+contract UpgradeCallBatches is UpgradeCallBatchLogic {
+    // entrypoint
+    function printCallBatches(
+        string memory _configFile,
+        string[] memory _domainNames,
+        string memory _localDomainName,
+        bool recovery
+    ) external {
+        localDomainName = _localDomainName;
+        setUp(_configFile);
+        for (uint256 i; i < _domainNames.length; i++) {
+            upgrade(_domainNames[i]);
+        }
+        writeCallBatch(recovery);
+    }
+
+    function setUp(string memory _configFile) internal {
+        __Config_initialize(_configFile);
+        string memory _outputFile = "upgradeActions.json";
+        __CallBatch_initialize(
+            localDomainName,
+            getDomainNumber(localDomainName),
+            _outputFile,
+            true
+        );
+        console2.log("Governance Actions have been output to", _outputFile);
```

Comment:
i tested implementing this and running it with the fork test and idk

Reply:
Since it's all serial, the output should be as well:
- Upgrading for Domain X
- Stage 1
- Stage 2
- Finish
- Upgrading for Domain Y

In multi-chain deployment, it was very helpful as an operator. A peace of mind that everything worked as expected. But I won't block or push for it.

Reply:
https://github.com/nomad-xyz/monorepo/pull/511/commits/d468099b2587cfbb8b0e0dac5985333c42ade07b